Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/WebrtcSignalControls.cs

```diff
@@ -0,0 +1,203 @@
+                {
+                    MessageType = SignalerMessage.WireMessageType.Answer,
+                    Data = answer,
+                    TargetId = TargetIdField.text
+                });
+            });
+
+            // bind our handler so when an ice message is ready we can to signalling
+            PeerEventsInstance.OnIceCandiateReadyToSend.AddListener((string candidate, int sdpMlineindex, string sdpMid) =>
+            {
+                signaler.SendMessageAsync(new SignalerMessage()
+                {
+                    MessageType = SignalerMessage.WireMessageType.Ice,
+                    Data = candidate + "|" + sdpMlineindex + "|" + sdpMid,
+                    IceDataSeparator = "|",
+                    TargetId = TargetIdField.text
+                });
+            });
+
+            signaler.OnMessage += (SignalerMessage msg) =>
+            {
+                // depending on what type of message we get, we'll handle it differently
+                // this is the "glue" that allows two peers to establish a connection.
+                switch (msg.MessageType)
+                {
+                    case SignalerMessage.WireMessageType.Offer:
+                        PeerEventsInstance.SetRemoteDescription("offer", msg.Data);
+                        // if we get an offer, we immediately send an answer
+                        PeerEventsInstance.CreateAnswer();
+                        break;
+                    case SignalerMessage.WireMessageType.Answer:
+                        PeerEventsInstance.SetRemoteDescription("answer", msg.Data);
+                        break;
+                    case SignalerMessage.WireMessageType.Ice:
+                        // this "parts" protocol is defined above, in PeerEventsInstance.OnIceCandiateReadyToSend listener
+                        var parts = msg.Data.Split(new string[] { msg.IceDataSeparator }, StringSplitOptions.RemoveEmptyEntries);
+                        PeerEventsInstance.AddIceCandidate(parts[0], int.Parse(parts[1]), parts[2]);
+                        break;
+                    case SignalerMessage.WireMessageType.SetPeer:
+                        // this allows a remote peer to set our text target peer id
+                        // it is primarily useful when one device does not support keyboard input
+                        //
+                        // note: when running this sample on hololens (for example) we may use postman or a similar
+                        // tool to use this message type to set the target peer. This is NOT a production-quality solution.
+                        TargetIdField.text = msg.Data;
+                        break;
+                    default:
+                        Debug.Log("Unknown message: " + msg.MessageType + ": " + msg.Data);
+                        break;
+                }
+            };
+
+            signaler.OnFailure += (Exception ex) =>
+            {
+                Debug.Log(ex.Message);
```

Comment:
This should update the SignalerStatusLabel instead of (or in addition to) logging.